lib/paperclip/storage/s3.rb

```diff
@@ -141,7 +141,7 @@ def sanitize_hash(hash)
             Proc.new do |style, attachment|
               permission  = (@s3_permissions[style.to_s.to_sym] || @s3_permissions[:default])
               permission  = permission.call(attachment, style) if permission.respond_to?(:call)
-              (permission == :public_read) ? 'http' : 'https'
+              (permission == :public_read) ? 'http'.freeze : 'https'.freeze
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.